zilencer/views.py

```diff
@@ -106,3 +114,66 @@ def remote_server_notify_push(request,  # type: HttpRequest
                                      badge=1, zulip=apns_payload)
 
     return json_success()
+
+def register_remote_server(request):
+    # type: (HttpRequest) -> HttpResponse
+    if request.method == 'POST':
+        form = RemoteServerRegistrationForm(request.POST)
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            api_key = form.cleaned_data['api_key']
+            uuid = form.cleaned_data['uuid']
+            hostname = form.cleaned_data['hostname']
+            with transaction.atomic():
+                status_obj = RemoteServerRegistrationStatus.objects.create(
+                    email=email,
+                    hostname=hostname,
+                    uuid=uuid,
+                    api_key=api_key,
+                )
+                template_prefix = 'zilencer/emails/confirm_remoteserver'
+                context = {'hostname': hostname}
+                RemoteServerConfirmation.objects.send_confirmation(
+                    status_obj,
+                    template_prefix,
+                    email,
+                    additional_context=context,
+                    host=request.get_host())
+                return HttpResponseRedirect(
+                    reverse('send_confirm', kwargs={'email': email}))
+    else:
+        form = RemoteServerRegistrationForm()
+
+    return render(request,
+                  'zilencer/register_remote_server.html',
+                  context={'form': form, 'current_url': request.get_full_path},
+                  )
+
+def confirm(request, confirmation_key):
+    # type: (HttpRequest, str) -> HttpResponse
+    confirmed = True
+    confirmation_key = confirmation_key.lower()
+    status_obj = RemoteServerConfirmation.objects.confirm(confirmation_key)
+    if not status_obj:
+        confirmed = False
+        try:
+            status_obj = RemoteServerConfirmation.objects.get(
+                confirmation_key=confirmation_key)
+        except RemoteServerConfirmation.DoesNotExist:
+            pass
+    else:
+        assert isinstance(status_obj, RemoteServerRegistrationStatus)
+        RemoteZulipServer.objects.create(
+            uuid=status_obj.uuid,
+            api_key=status_obj.api_key,
+            hostname=status_obj.hostname,
+            contact_email=status_obj.email,
+        )
+
+    ctx = {
+        'confirmed': confirmed,
+        'support_email': settings.ZULIP_ADMINISTRATOR,
```

Comment:
I do have one concern here though. In the context processor, we have the email address of the Zulip administration and maybe in the future we would want to keep the admin email and support emails separate.